File: Everything/ClientApp/src/models/budget/IncomeSource.ts
export default class IncomeSource {
    id: number = 0;
    name: string | null = null;
    description: string = '';
    amount: number = 0;
    budgetId: number = 0;

    constructor(init?: Partial<IncomeSource>) {
        Object.assign(this, init);
    }
}

File: Everything/Models/Budget/IncomeSource.cs
using System.Collections.Generic;

namespace everything.Models
{
    public class IncomeSource
    {
        public IncomeSource()
        {
            Deductions = new List<IncomeSourceDeduction>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Amount { get; set; }

        public int BudgetId { get; set; }
        public virtual Budget Budget { get; set; }

        //thinking this will only be used for savings accounts?
        public int? DepositAccountId { get; set; }
        public virtual Account DepositAccount { get; set; }

        public virtual IEnumerable<IncomeSourceDeduction> Deductions { get; set; }

        // TODO: freqency?
    }
}
